### PR TITLE
Ancillary mask feature

### PR DESCRIPTION
**Pull request by cssprad1**
# MODIS Water Simple Algorithm and Post Processing Update

## Description
Differences were noticed when comparing the C6 water masks with water masks generated by this application using the simple algorithm. This facilitated the need for updating some parts of the simple algorithm's decision tree and adding post-processing layers which were used in C6 data generation. The addition of more post-processing added the need for combining these products into a single packed-bit mask.

### Simple Algorithm Decision Tree update
- Scaled NDVI values to match the NDVI data range
- Changed code to where any pixel that summed B1 and B2 to 0, be bad data because of divide by zero error when calculating NDVI

### Pre-processing QA mask
- Changed QA of SR data to match C6 code base
	- Splitting QA to only apply based on labeled water pixel value

### Post-processing QA
- Updated QA process to account for all new products. Followed the rules which were written in the MODIS Water document AND the C6 code base. 
- Changed the view to only accept a single post-processing mask which is in a packed-bit int16 format. 
- Post-processing mask is available for all 318 tiles
- This mask contains
	- Impervious masking
	- Permanent water masking
	- Ancillary data masking
	- GMTED DEM slope masking
	- Static seven class 

## Testing

- Added/updated unit tests
	- MaskGenerator unit test
	- QA unit test

- [x] Code passes PEP 8 (with exceptions in code base)
- [x] Unit tests pass

### CODE REVIEW

**Review by evhrgh — APPROVED** [review]
I added two minor comments.  Github let me connect them to the specific lines to which they referred.  I will assume Github will alert you to them somehow.

I like doing this through Github.

File: model/MaskGenerator.py [hunk 1]
```diff
@@ -43,86 +43,75 @@ def __init__(self, bandDict):
...
+            self._printGeneralMaskDebugInfo()
 
-            (self._bandDict[br.STATE] & MaskGenerator.CLOUD_SHADOW ==
-                MaskGenerator.CLOUD_SHADOW) |
+        return mask
 
-            (self._bandDict[br.STATE] & MaskGenerator.CLOUD_INT ==
-                MaskGenerator.CLOUD_INT),
+    # -------------------------------------------------------------------------
+    # generateMask
```
[the remark this answers is not in the document]
Comment (cssprad1) [reply model/MaskGenerator.py]: Fixed in commit: https://github.com/nasa-nccs-hpda/modis_water/pull/16/commits/266b2a268c1ed6ade018b5e850c98ba8ad47e249

Thanks

File: model/Utils.py [hunk 2]
```diff
@@ -78,3 +79,18 @@ def getPostStr():
...
         return post_str
+
+    # -------------------------------------------------------------------------
+    # getStaticDatasetPath
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def getStaticDatasetPath(inDir, searchTerm) -> str:
+        searchPath = os.path.join(
+            inDir, searchTerm)
+        staticPath = glob.glob(searchPath)
```
[the remark this answers is not in the document]
Comment (cssprad1) [reply model/Utils.py:90]: Good to know. Yes I definitely need to make the switch from os+glob to pathlib. It seems like it has a lot more functionality and is what most people are using these days. 

Thanks!

**Comment by cssprad1** [comment]
Fixed requests from review.

**Pull request merged**